tests/test_13_validate.py

```diff
@@ -146,3 +147,61 @@ def test_valid_address():
         assert valid_address("[2001:8003:5555:9999:555a:5555:c77:d5c5")
     with raises(NotValid):
         assert valid_address("[[2001:8003:5555:9999:555a:5555:c77:d5c5]")
+
+
+def test_valid_domain_name():
+    assert valid_domain_name("api.my-domain.com")
+    assert valid_domain_name("auth.admin.domain.com")
+    assert valid_domain_name("auth.domain.com")
+    assert valid_domain_name("auth.domain.com")
+    assert valid_domain_name("lk.domain.com:12")
+    assert valid_domain_name("lk.domain.com:12")
+    assert valid_domain_name("static.domain.xyz:12345")
+    assert valid_domain_name("domain.com")
+    assert valid_domain_name("domain.lu")
+    assert valid_domain_name("auth-domain.com")
+    assert valid_domain_name("domain.com:12345")
+    assert valid_domain_name("auth-admin.domain-uero.xyz")
+    assert valid_domain_name("auth.lk.d.sr.mydomain.com")
+
+    with raises(ValueError):
+        valid_domain_name("")
+
+    with raises(ValueError):
+        valid_domain_name("auth.domain.ljnjnfds")
```

Comment:
Why should this result to an error?

Reply:
top-level domain cannot be longer than 5 characters

Reply:
> top-level domain cannot be longer than 5 characters

According to the [MDN](https://developer.mozilla.org/en-US/docs/Learn/Common_questions/Web_mechanics/What_is_a_domain_name), the longest a TLD can be is 63 characters. Cutting this down to a 5-character space would invalidate many top level domains, some that I own, some that I know others own.